block/manager.go

```diff
@@ -504,7 +504,7 @@ func (m *Manager) publishBlock(ctx context.Context) error {
 		proposer := &tmtypes.Validator{Address: pubKey.Address(), PubKey: pubKey}
 		// TODO: read staking query to construct validators
 		block.SignedHeader.Validators = &tmtypes.ValidatorSet{
-			Validators: []*tmtypes.Validator{proposer},
+			Validators: m.lastState.Validators.Validators,
 			Proposer:   proposer,
```

Comment:
ideally we should have the signer being one of the validators (genesis validators in the tests).
instead of reseting the `Proposer` to signing aggregator, can you make sure that signing key corresponds to the validator?

for example, you could do this in your test:
```
nodeKey := &p2p.NodeKey{
      PrivKey: vKeys[0],
}
signingKey, _ := conv.GetNodeKey(nodeKey)
```

Reply:
Added a helper `getGenesisValidatorSetWithSigner` which returns both validator set and signer.